src/ArraySet/ArraySet.spec.ts

```diff
@@ -0,0 +1,267 @@
+﻿import { describe, expect, it } from "vitest"
+import { ArraySet } from "../ArraySet"
+import type { EqualityComparer } from "../EqualityComparer"
+import { pipe } from "../Composition"
+
+const caseInsensitiveEqualityComparer: EqualityComparer<string> = {
+    equals: (s1, s2) => s1.toLowerCase() === s2.toLowerCase(),
+}
+
+describe("ArraySet", () => {
```

Comment:
💭 Wondering if type tests would be useful here? Just essentially verifying all the functions  return an actual `ArraySet` type.

Reply:
I'll take a look-see!